.github/workflows/codeql-analysis.yml

```diff
@@ -33,6 +35,15 @@ jobs:
           distribution: 'zulu'
           java-version: '11'
 
+      - name: copy maven settings
+        run: |
+          cp $HOME/.m2/settings.xml /tmp/settings.xml
+
+      - name: Modify settings.xml
+        if: ${{ env.USE_STAGE == 'true' }}
+        run: |
+          sed -i 's/<\/settings>/<profiles><profile><id>local-repo<\/id><repositories><repository><id>central<\/id><url>https:\/\/repo.maven.apache.org\/maven2<\/url><releases><enabled>true<\/enabled><\/releases><snapshots><enabled>false<\/enabled><\/snapshots><\/repository><repository><id>staged-releases<\/id><url>https:\/\/repository.apache.org\/content\/groups\/staging\/<\/url><\/repository><\/repositories><pluginRepositories><pluginRepository><id>staged-releases<\/id><url>https:\/\/repository.apache.org\/content\/groups\/staging\/<\/url><\/pluginRepository><\/pluginRepositories><\/profile><\/profiles><activeProfiles><activeProfile>local-repo<\/activeProfile><\/activeProfiles><\/settings>/' $HOME/.m2/settings.xml
```

Comment:
or maybe we should add the `stage-version` settings.xml in the `.github/configs/settings.xml` & override it directly? (to avoid hard-code everywhere